src/main/java/org/craftercms/engine/store/s3/S3SiteListResolver.java

```diff
@@ -58,24 +60,29 @@ public void setSiteNameMacroName(String siteNameMacroName) {
 
     @Override
     public Collection<String> getSiteList() {
-        String bucketName = s3Uri.getBucket();
+        String bucketName = s3Uri.bucket().orElse("");
```

Comment:
In what scenarios is an empty bucket name a valid configuration? Should we throw an exception here?

Reply:
The bucket will never be empty. It will throw an exception when parsing S3 URI if the format is incorrect. However, I switched this to `orElseThrow()`, which should never happen.